src/Sarif.Viewer.VisualStudio/CodeAnalysisResultManager.cs

```diff
@@ -557,5 +558,44 @@ private void RemoveTemporaryFiles()
             // Unit tests will only create one cache.
             return this.RunIndexToRunDataCache.Values.First().RemappedPathPrefixes.ToArray();
         }
+
+        // Extract selected results from original SarifLog.
+        internal SarifLog GetPartitionedLog(IEnumerable<SarifErrorListItem> listItems)
+        {
+            if (listItems.Any())
+            {
+                int runIndex = listItems.First().RunIndex;
+                string guid = Guid.NewGuid().ToString();
+                foreach (SarifErrorListItem item in listItems)
+                {
+                    if (item.SarifResult != null)
+                    {
+                        item.SarifResult.Guid = guid;
+                    }
+                }
+
+                if (listItems.Any(i => i.SarifResult != null))
+                {
+                    // parition results in log
+                    PartitionFunction<string> partitionFunction = (result) => result.Guid ??= Guid.NewGuid().ToString();
```

Comment:
>PartitionFunction<string> partitionFunction = (result) => result.Guid ??= Guid.NewGuid().ToString(); [](start = 20, length = 100)

try something like: result.Guid != null ? result.Guid : null. With that the partition would ignore the data

Reply:
done